kratos/includes/kratos_components.h

```diff
@@ -76,31 +73,28 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
     ///@{
 
 
-
     ///@}
     ///@name Operations
     ///@{
 
-
     static void Add(std::string const& Name, TComponentType const& ThisComponent)
     {
-        msComponents.insert(typename ComponentsContainerType::value_type(Name , &ThisComponent));
+        msComponents.insert(ValueType(Name , &ThisComponent));
     }
 
-
-//     static void Add(std::string const& Name, TComponentType const& ThisComponent, ComponentsContainerType& ThisComponents)
-//     {
-    ////ThisComponents.insert(typename ComponentsContainerType::value_type(Name ,boost::cref(ThisComponent)));
-    //msComponents.insert(typename ComponentsContainerType::value_type(Name ,boost::cref(ThisComponent)));
-//     }
-
-
     static TComponentType const& Get(std::string const& Name)
     {
-        typename ComponentsContainerType::iterator i =  msComponents.find(Name);
-        if(i == msComponents.end())
-          KRATOS_THROW_ERROR(std::invalid_argument, "The component is not registered!", Name);
-        return *(i->second);
+        auto it_comp =  msComponents.find(Name);
+        if (it_comp != msComponents.end()) {
+            return *(it_comp->second);
+        }
+
+        std::stringstream err_msg;
+        err_msg << "The component \"" << Name << "\" is not registered!\n"
+                << "Maybe you need to import the application where it is defined?\n"
+                << "The following components of this type are registered:" << std::endl;
+        PrintData(err_msg);
```

Comment:
@jcotela I use it here, thats why it has to be static

=> after you pointed out that this is problematic I will revert making the print-fcts static and just create an instance of `KratosComponents` here for using it

Reply:
As you see, it is not a problem either way, but it is something we need to be aware of. Unfortunately, I could not find what is the equivalent of `__str__` for `print(<class>)` (as opposed to `print(<instance>)`.

Reply:
but we can export it to python as a normal method. Isn't it?

Reply:
Ofc you can export it, but then the „PrintObject“ - fct I recently added cannot be used

Either way I solved this another way,  no modifications to the Print-fcts are necessary now

Reply:
How you solved it?

Reply:
By creating an instance:
https://github.com/KratosMultiphysics/Kratos/blob/07c074218bd3dbd4ed2027fe6248574162b7d593/kratos/includes/kratos_components.h#L96-L97

Reply:
I think this solution is fine since anyway an error is thrown

Reply:
is fine.....
Nevertheless having a private static `PrintDataImplementation` would be a cleaner....

Reply:
should I change it?

If not then this is ready for merge

Reply:
It is fine as it is as we should replace it in future with registry